godot-core/src/builtin/vectors/vector_macros.rs

```diff
@@ -484,16 +484,17 @@ macro_rules! impl_vector_fns {
             #[inline]
             pub fn sign(self) -> Self {
                 #[inline]
-                fn f(x: i32) -> i32 {
-                    match x.cmp(&0) {
-                        Ordering::Equal => 0,
-                        Ordering::Greater => 1,
-                        Ordering::Less => -1,
+                fn f(x: $Scalar) -> $Scalar {
+                    let r = x.partial_cmp(&(0 as $Scalar)).unwrap_or_else(|| panic!("Vector component {x} isn't signed!"));
+                    match r {
+                        Ordering::Equal => 0 as $Scalar,
+                        Ordering::Greater => 1 as $Scalar,
+                        Ordering::Less => -1 as $Scalar,
```

Comment:
Maybe while at it, rename `x` to `c` for component. It's a bit misleading otherwise.